hypervisor/arch/x86/guest/vlapic.c

```diff
@@ -1936,6 +1936,32 @@ static void vlapic_timer_expired(void *data)
 	}
 }
 
+static inline bool is_x2apic_enabled(const struct acrn_vlapic *vlapic)
+{
+	if ((vlapic_get_apicbase(vlapic) & APICBASE_X2APIC) == 0UL) {
+		return false;
+	} else {
+		return true;
+	}
+}
+
+static int vlapic_x2apic_access(struct vcpu *vcpu)
+{
+	struct acrn_vlapic *vlapic;
+	int error = 0;
+
+	vlapic = vcpu_vlapic(vcpu);
+	if (is_x2apic_enabled(vlapic) == false) {
```

Comment:
if (!is_x2apic_enabled())   ?

Reply:
Remember we need to use == false, to follow some MISRA-C rule